src/library_wasmfs.js

```diff
@@ -190,9 +190,16 @@ FS.createPreloadedFile = FS_createPreloadedFile;
       return bytesRead;
     },
     // TODO: allocate
-    // TODO: mmap
-    // TODO: msync
-    // TODO: munmap
+    mmap: (stream, length, position, prot, flags) => {
+      var buf = FS.handleError(__wasmfs_mmap(length, prot, flags, stream.fd, position));
+      return { ptr: buf, allocated: true };
+    },
+    msync: (stream, bufferPtr, offset, length, mmapFlags) => {
+      return FS.handleError(__wasmfs_msync(bufferPtr, length, mmapFlags));
```

Comment:
Perhaps we should assert on the unused parameters not being used? And comment why they are here (backwards compatibility?)

Reply:
What would be the best way to assert that `offset` is not used? One idea I had was to `assert(offset == 0)`, since `offset` has to have some value so that `length` and `mmapFlags` get passed correctly. Is there a better way to do this?

Reply:
Yes, that's what I had in mind for `offset`.

I'm less sure about `stream`. Perhaps just add a TODO to validate that the stream has the right fd for the mapping (`emscripten_mmap.c` can do that).